packages/cdktf-cli/bin/cmds/helper/outputs.ts

```diff
@@ -0,0 +1,47 @@
+import * as fs from "fs";
+import * as path from "path";
+import { TerraformOutput, isTerraformOutput } from "../ui/models/terraform";
+import { NestedTerraformOutput } from "../ui/terraform-context";
+
+export type Outputs = { [key: string]: TerraformOutput };
+export type OutputIdMapLeaf = { [constructId: string]: string };
+export type OutputIdMapNode = { [stackOrConstructId: string]: OutputIdMap };
+export type OutputIdMap = OutputIdMapLeaf | OutputIdMapNode;
+
+function unpackTerraformOutput(
+  outputs: NestedTerraformOutput,
+  includeSensitiveOutputs: boolean
+): Record<string, string> {
+  return Object.entries(outputs).reduce(
+    (acc, [key, entry]) => ({
+      ...acc,
+      [key]: isTerraformOutput(entry)
+        ? !entry.sensitive || includeSensitiveOutputs
+          ? entry.value
+          : "<sensitive>"
+        : unpackTerraformOutput(entry, includeSensitiveOutputs),
+    }),
+    {}
+  );
+}
+
+export async function saveOutputs(
+  filePath: string,
+  outputs: NestedTerraformOutput,
+  includeSensitiveOutputs: boolean
+) {
+  fs.writeFileSync(
+    filePath,
+    JSON.stringify(
+      unpackTerraformOutput(outputs, includeSensitiveOutputs),
+      null,
+      2
+    )
+  );
+}
+
+export function normalizeOutputPath(filePath: string) {
```

Comment:
Would it make sense to also automatically add a `.json` suffix if the passed filename does not end with `.json` already? I have no strong opinion on this, just curious what you think.

Reply:
I'd vote for predictability and therefore not add anything :)

Reply:
Sounds good 👍